In [20]:
import pandas as pd

#### For year 2012

In [21]:
data = pd.read_csv("./Poverty/ACS_12_5YR_S1701/ACS_12_5YR_S1701_with_ann.csv",encoding='ISO-8859-1',low_memory=False)

In [22]:
for i in data.iloc[0]:
    print(i)

Id
Id2
Geography
Total; Estimate; Population for whom poverty status is determined
Total; Margin of Error; Population for whom poverty status is determined
Below poverty level; Estimate; Population for whom poverty status is determined
Below poverty level; Margin of Error; Population for whom poverty status is determined
Percent below poverty level; Estimate; Population for whom poverty status is determined
Percent below poverty level; Margin of Error; Population for whom poverty status is determined
Total; Estimate; AGE - Under 18 years
Total; Margin of Error; AGE - Under 18 years
Below poverty level; Estimate; AGE - Under 18 years
Below poverty level; Margin of Error; AGE - Under 18 years
Percent below poverty level; Estimate; AGE - Under 18 years
Percent below poverty level; Margin of Error; AGE - Under 18 years
Total; Estimate; AGE - Under 18 years - Related children under 18 years
Total; Margin of Error; AGE - Under 18 years - Related children under 18 years
Below poverty level; E

In [23]:
data.columns

Index(['GEO.id', 'GEO.id2', 'GEO.display-label', 'HC01_EST_VC01',
       'HC01_MOE_VC01', 'HC02_EST_VC01', 'HC02_MOE_VC01', 'HC03_EST_VC01',
       'HC03_MOE_VC01', 'HC01_EST_VC03',
       ...
       'HC02_EST_VC65', 'HC02_MOE_VC65', 'HC03_EST_VC65', 'HC03_MOE_VC65',
       'HC01_EST_VC68', 'HC01_MOE_VC68', 'HC02_EST_VC68', 'HC02_MOE_VC68',
       'HC03_EST_VC68', 'HC03_MOE_VC68'],
      dtype='object', length=279)

In [24]:
gender = data[['GEO.id2','HC01_EST_VC09','HC02_EST_VC09','HC03_EST_VC09']]

In [25]:
data.head(5)

GEO.id GEO.id2        GEO.display-label  \
0              Id     Id2                Geography   
1  0500000US01001   01001  Autauga County, Alabama   
2  0500000US01003   01003  Baldwin County, Alabama   
3  0500000US01005   01005  Barbour County, Alabama   
4  0500000US01007   01007     Bibb County, Alabama   

                                       HC01_EST_VC01  \
0  Total; Estimate; Population for whom poverty s...   
1                                              54255   
2                                             180970   
3                                              24530   
4                                              21794   

                                       HC01_MOE_VC01  \
0  Total; Margin of Error; Population for whom po...   
1                                                116   
2                                                435   
3                                                396   
4                                                170   

                                       HC02_EST_VC01  \
0  Below poverty level; Estimate; Population for ...   
1                                               6267   
2                                              24002   
3                                               6395   
4                                               3594   

                                       HC02_MOE_VC01  \
0  Below poverty level; Margin of Error; Populati...   
1                                                879   
2                                               2188   
3                                                650   
4                                                733   

                                       HC03_EST_VC01  \
0  Percent below poverty level; Estimate; Populat...   
1                                               11.6   
2                                               13.3   
3                                               26.1   
4                                               16.5   

                                       HC03_MOE_VC01  \
0  Percent below poverty level; Margin of Error; ...   
1                                                1.6   
2                                                1.2   
3                                                2.7   
4                                                3.4   

                           HC01_EST_VC03  ...  \
0  Total; Estimate; AGE - Under 18 years  ...   
1                                  14611  ...   
2                                  41931  ...   
3                                   5988  ...   
4                                   5253  ...   

                                 HC02_EST_VC65  \
0  Below poverty level; Estimate; Did not work   
1                                         1469   
2                                         3413   
3                                         1160   
4                                          562   

                                       HC02_MOE_VC65  \
0  Below poverty level; Margin of Error; Did not ...   
1                                                255   
2                                                434   
3                                                182   
4                                                165   

                                       HC03_EST_VC65  \
0  Percent below poverty level; Estimate; Did not...   
1                                               47.6   
2                                               33.8   
3                                               58.7   
4                                               42.6   

                                       HC03_MOE_VC65  \
0  Percent below poverty level; Margin of Error; ...   
1                                                6.0   
2                                                3.4   
3                                                5.4   
4                                                9.5   

                                       HC01_EST_VC68  \
0

In [26]:
data = data.iloc[:, : 9]
data.columns

Index(['GEO.id', 'GEO.id2', 'GEO.display-label', 'HC01_EST_VC01',
       'HC01_MOE_VC01', 'HC02_EST_VC01', 'HC02_MOE_VC01', 'HC03_EST_VC01',
       'HC03_MOE_VC01'],
      dtype='object')

In [27]:
data = data[['GEO.id2', 'GEO.display-label', 'HC01_EST_VC01', 'HC02_EST_VC01','HC03_EST_VC01']]
data.columns = data.iloc[0]
data = data.drop(data.index[0])
data.head(5)

0    Id2                Geography  \
1  01001  Autauga County, Alabama   
2  01003  Baldwin County, Alabama   
3  01005  Barbour County, Alabama   
4  01007     Bibb County, Alabama   
5  01009   Blount County, Alabama   

0 Total; Estimate; Population for whom poverty status is determined  \
1                                              54255                  
2                                             180970                  
3                                              24530                  
4                                              21794                  
5                                              56993                  

0 Below poverty level; Estimate; Population for whom poverty status is determined  \
1                                               6267                                
2                                              24002                                
3                                               6395                                
4                                               3594                                
5                                               8396                                

0 Percent below poverty level; Estimate; Population for whom poverty status is determined  
1                                               11.6                                       
2                                               13.3                                       
3                                               26.1                                       
4                                               16.5                                       
5                                               14.7

In [28]:
data.columns

Index(['Id2', 'Geography',
       'Total; Estimate; Population for whom poverty status is determined',
       'Below poverty level; Estimate; Population for whom poverty status is determined',
       'Percent below poverty level; Estimate; Population for whom poverty status is determined'],
      dtype='object', name=0)

In [29]:
data["Total; Estimate; Population for whom poverty status is determined"] = pd.to_numeric(data["Total; Estimate; Population for whom poverty status is determined"])
data["Below poverty level; Estimate; Population for whom poverty status is determined"] = pd.to_numeric(data["Below poverty level; Estimate; Population for whom poverty status is determined"])
data["Percent below poverty level; Estimate; Population for whom poverty status is determined"] = pd.to_numeric(data["Percent below poverty level; Estimate; Population for whom poverty status is determined"])
data.shape

(3143, 5)

In [30]:
data["Calculated_Population"] = (data["Below poverty level; Estimate; Population for whom poverty status is determined"]
                                 /data["Percent below poverty level; Estimate; Population for whom poverty status is determined"])*100

In [31]:
data.head(5)

0    Id2                Geography  \
1  01001  Autauga County, Alabama   
2  01003  Baldwin County, Alabama   
3  01005  Barbour County, Alabama   
4  01007     Bibb County, Alabama   
5  01009   Blount County, Alabama   

0  Total; Estimate; Population for whom poverty status is determined  \
1                                              54255                   
2                                             180970                   
3                                              24530                   
4                                              21794                   
5                                              56993                   

0  Below poverty level; Estimate; Population for whom poverty status is determined  \
1                                               6267                                 
2                                              24002                                 
3                                               6395                                 
4                                               3594                                 
5                                               8396                                 

0  Percent below poverty level; Estimate; Population for whom poverty status is determined  \
1                                               11.6                                         
2                                               13.3                                         
3                                               26.1                                         
4                                               16.5                                         
5                                               14.7                                         

0  Calculated_Population  
1           54025.862069  
2          180466.165414  
3           24501.915709  
4           21781.818182  
5           57115.646259

In [32]:
data['is_calculated_pop_greater'] = (data['Total; Estimate; Population for whom poverty status is determined'] < data['Calculated_Population']).astype(bool)

In [33]:
data.head(5)

0    Id2                Geography  \
1  01001  Autauga County, Alabama   
2  01003  Baldwin County, Alabama   
3  01005  Barbour County, Alabama   
4  01007     Bibb County, Alabama   
5  01009   Blount County, Alabama   

0  Total; Estimate; Population for whom poverty status is determined  \
1                                              54255                   
2                                             180970                   
3                                              24530                   
4                                              21794                   
5                                              56993                   

0  Below poverty level; Estimate; Population for whom poverty status is determined  \
1                                               6267                                 
2                                              24002                                 
3                                               6395                                 
4                                               3594                                 
5                                               8396                                 

0  Percent below poverty level; Estimate; Population for whom poverty status is determined  \
1                                               11.6                                         
2                                               13.3                                         
3                                               26.1                                         
4                                               16.5                                         
5                                               14.7                                         

0  Calculated_Population  is_calculated_pop_greater  
1           54025.862069                      False  
2          180466.165414                      False  
3           24501.915709                      False  
4           21781.818182                      False  
5           57115.646259                       True

In [34]:
print("Count where calculated population is greater:")
data.is_calculated_pop_greater.astype(bool).sum(axis=0)

Count where calculated population is greater:


1525

In [35]:
print("Count where calculated population is lesser:")
zero = data[data['is_calculated_pop_greater'] == False]
zero.shape[0]

Count where calculated population is lesser:


1618

#### ------------------- Champaign population 2012 ---------------------

In [55]:
data[data['Id2'] == '17019']

0      Id2                   Geography  \
605  17019  Champaign County, Illinois   

0    Total; Estimate; Population for whom poverty status is determined  \
605                                             184223                   

0    Below poverty level; Estimate; Population for whom poverty status is determined  \
605                                              40685                                 

0    Percent below poverty level; Estimate; Population for whom poverty status is determined  \
605                                               22.1                                         

0    Calculated_Population  is_calculated_pop_greater  
605          184095.022624                      False

#### Actual population: 204099
#### Pop from Rohan's data: 184223
#### Pop from WNV data: 203937
#### -----------------------------------------------------------------------------

#### Gender Analysis

In [36]:
print(gender.shape)
gender.head(5)

(3144, 4)


GEO.id2                HC01_EST_VC09  \
0     Id2  Total; Estimate; SEX - Male   
1   01001                        26332   
2   01003                        87695   
3   01005                        11923   
4   01007                        11290   

                               HC02_EST_VC09  \
0  Below poverty level; Estimate; SEX - Male   
1                                       2663   
2                                      10784   
3                                       2755   
4                                       1688   

                                       HC03_EST_VC09  
0  Percent below poverty level; Estimate; SEX - Male  
1                                               10.1  
2                                               12.3  
3                                               23.1  
4                                               15.0

In [37]:
gender.columns = gender.iloc[0]
gender = gender.drop(data.index[0])
gender.head(5)

0    Id2  Total; Estimate; SEX - Male  \
0    Id2  Total; Estimate; SEX - Male   
2  01003                        87695   
3  01005                        11923   
4  01007                        11290   
5  01009                        28290   

0  Below poverty level; Estimate; SEX - Male  \
0  Below poverty level; Estimate; SEX - Male   
2                                      10784   
3                                       2755   
4                                       1688   
5                                       3714   

0  Percent below poverty level; Estimate; SEX - Male  
0  Percent below poverty level; Estimate; SEX - Male  
2                                               12.3  
3                                               23.1  
4                                               15.0  
5                                               13.1

In [39]:
gender.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3143 entries, 0 to 3143
Data columns (total 4 columns):
Id2                                                  3143 non-null object
Total; Estimate; SEX - Male                          3143 non-null object
Below poverty level; Estimate; SEX - Male            3143 non-null object
Percent below poverty level; Estimate; SEX - Male    3143 non-null object
dtypes: object(4)
memory usage: 122.8+ KB


In [44]:
gender.iloc[0]

0
Id2                                                                                                Id2
Total; Estimate; SEX - Male                                                Total; Estimate; SEX - Male
Below poverty level; Estimate; SEX - Male                    Below poverty level; Estimate; SEX - Male
Percent below poverty level; Estimate; SEX - Male    Percent below poverty level; Estimate; SEX - Male
Name: 0, dtype: object

In [45]:
gender.columns = gender.iloc[0]
gender = gender.drop(gender.index[0])

In [46]:
gender["Total; Estimate; SEX - Male"] = pd.to_numeric(gender["Total; Estimate; SEX - Male"])
gender["Below poverty level; Estimate; SEX - Male"] = pd.to_numeric(gender["Below poverty level; Estimate; SEX - Male"])
gender["Percent below poverty level; Estimate; SEX - Male"] = pd.to_numeric(gender["Percent below poverty level; Estimate; SEX - Male"])

In [48]:
gender.head(5)

0    Id2  Total; Estimate; SEX - Male  \
2  01003                        87695   
3  01005                        11923   
4  01007                        11290   
5  01009                        28290   
6  01011                         5601   

0  Below poverty level; Estimate; SEX - Male  \
2                                      10784   
3                                       2755   
4                                       1688   
5                                       3714   
6                                       1079   

0  Percent below poverty level; Estimate; SEX - Male  
2                                               12.3  
3                                               23.1  
4                                               15.0  
5                                               13.1  
6                                               19.3

In [49]:
gender["Calculated_Population"] = (gender["Below poverty level; Estimate; SEX - Male"]
                                 /gender["Percent below poverty level; Estimate; SEX - Male"])*100
gender.head(5)

0    Id2  Total; Estimate; SEX - Male  \
2  01003                        87695   
3  01005                        11923   
4  01007                        11290   
5  01009                        28290   
6  01011                         5601   

0  Below poverty level; Estimate; SEX - Male  \
2                                      10784   
3                                       2755   
4                                       1688   
5                                       3714   
6                                       1079   

0  Percent below poverty level; Estimate; SEX - Male  Calculated_Population  
2                                               12.3           87674.796748  
3                                               23.1           11926.406926  
4                                               15.0           11253.333333  
5                                               13.1           28351.145038  
6                                               19.3            5590.673575

In [50]:
gender['is_calculated_pop_greater'] = (gender['Total; Estimate; SEX - Male'] < gender['Calculated_Population']).astype(bool)
gender.head(5)

0    Id2  Total; Estimate; SEX - Male  \
2  01003                        87695   
3  01005                        11923   
4  01007                        11290   
5  01009                        28290   
6  01011                         5601   

0  Below poverty level; Estimate; SEX - Male  \
2                                      10784   
3                                       2755   
4                                       1688   
5                                       3714   
6                                       1079   

0  Percent below poverty level; Estimate; SEX - Male  Calculated_Population  \
2                                               12.3           87674.796748   
3                                               23.1           11926.406926   
4                                               15.0           11253.333333   
5                                               13.1           28351.145038   
6                                               19.3            5590.673575   

0  is_calculated_pop_greater  
2                      False  
3                       True  
4                      False  
5                       True  
6                      False

In [51]:
print("Count where calculated population is greater:")
gender.is_calculated_pop_greater.astype(bool).sum(axis=0)

Count where calculated population is greater:


1567

In [52]:
print("Count where calculated population is lesser:")
rem = gender[gender['is_calculated_pop_greater'] == False]
rem.shape[0]

Count where calculated population is lesser:


1575

In [54]:
gender[gender['Id2'] == '17019']

0      Id2  Total; Estimate; SEX - Male  \
605  17019                        91622   

0    Below poverty level; Estimate; SEX - Male  \
605                                      19228   

0    Percent below poverty level; Estimate; SEX - Male  Calculated_Population  \
605                                               21.0           91561.904762   

0    is_calculated_pop_greater  
605                      False